### PR TITLE
Implement User logout and token Black listing

### PR DESCRIPTION
**Pull request by meshnesh**
# What does this PR do?
* Implement logout of the user from the application and token black listing

## Description of tasks to be completed

* Write tests for user logout and token blacklisting
* Create for black listed tokens

## How should this be tested manually?

* Run the app login to the application, which generates a token.
* Use the generated token to create and event.
* Logout from the application then try and create another event with the previous generated token

### CODE REVIEW

**Review by Rodgers-M — COMMENTED** [review]
Good implementation. Just curious, why blacklist a token when logging out while it will finally expire and become invalid?

**Comment by meshnesh** [comment]
@Rodgers-M  The token is blacklisted only if they log out before it expires so that if they try and log in again the token will be invalid.

File: app/models.py [hunk 1]
```diff
@@ -39,6 +39,42 @@ def save(self):
         db.session.add(self)
         db.session.commit()
 
+    def generate_token(self, user_id):
+        """ Generates the access token"""
+
+        try:
+            # set up a payload with an expiration time
+            payload = {
+                'exp': datetime.utcnow() + timedelta(minutes=1440),
```
Comment (frankip) [inline app/models.py]: don't you think this a lot of time for a session to be active?

File: app/auth/views.py [hunk 2]
```diff
@@ -80,6 +80,28 @@ def post(self):
...
             return make_response(jsonify(response)), 500
 
+class RestEmailView(MethodView):
+    """This class resets a users password."""
+
+    def post(self):
+        """Handle PUT request for this view. Url ---> /auth/register"""
+
+        # Query to see if the user email exists
+        user = User.query.filter_by(email=request.data['email']).first()
```
Comment (frankip) [inline app/auth/views.py]: good implementation

**Review by frankip — APPROVED** [review]
Thumbs up on the test and the logout implementation

File: app/models.py (hunk 1 above)
Comment (meshnesh) [reply app/models.py]: it's validity is 24hours, although I changed it to 12 hours, to avoid the user logging in every few minutes

**Pull request merged**